components/utils/src/templates.rs

```diff
@@ -78,6 +63,32 @@ pub fn rewrite_theme_paths(tera_theme: &mut Tera, theme: &str) {
     tera_theme.templates.extend(new_templates);
 }
 
+/// Checks for the presence of a given template. If none is found, also looks for a 
+/// fallback in theme and default templates. Returns the path of the most specific
+/// template found, or none if none are present.
+pub fn check_template_fallbacks(name: &str, tera: &Tera, theme: &Option<String>) -> Option<String> {
```

Comment:
can you return a `&str` instead?

Reply:
I'd tried doing that initially, and after wrestling with the borrow checker gave up. Clearly, trying to make a pull request when ill is not a good idea as it worked much better this morning.

Thank you for reminding me, commit just pushed now.